core/src/test/scala/org/apache/spark/rdd/PairRDDFunctionsSuite.scala

```diff
@@ -739,7 +739,7 @@ class PairRDDFunctionsSuite extends SparkFunSuite with SharedSparkContext {
         val dist = new BinomialDistribution(trials, p)
         val q = dist.cumulativeProbability(actual)
         withClue(s"p = $p: trials = $trials") {
-          assert(q >= 0.001 && q <= 0.999)
+          assert(q >= 0.001 && q < 0.99999)
```

Comment:
Otherwise the assert fails with:
```
p = 0.1: trials = 30 0.9995456448850946 was greater than or equal to 0.001, but 0.9995456448850946 was not less than or equal to 0.999
```
I will change the assert to `assert(0.0 <= q && q <= 1.0)`

Reply:
It's OK to increase the bounds, I was just saying they should be symmetric. Asserting it's in [0,1] is barely asserting anything except that the value isn't completely wrong. I'm OK with that too, but, don't think we should ever see 0 or 1. (0,1) could be a little more reasonable.

Reply:
> don't think we should ever see 0 or 1. (0,1) could be a little more reasonable.

ok. I will replace it by `assert(0.0 < q && q < 1.0)`